lua/neo-tree/ui/renderer.lua

```diff
@@ -933,6 +922,22 @@ M.is_window_valid = function(winid)
   end
 end
 
+M.update_floating_window_layouts = function()
+  for i, win in ipairs(floating_windows) do
+    --first remove any invalid windows
+    if M.is_window_valid(win.winid) == false then
+      table.remove(floating_windows, i)
+    else
```

Comment:
Removing values from the middle of a table you are iterating on is a bad idea and could produce bugs. In practice we don't usually have more than one window in this collection so you may not have noticed any bugs in testing, but one day this could come back to bite somebody.

It's also generally a good idea to use truthy assertions rather than negative ones where possible, because it is a little easier to read them.

I would suggest:
```suggestion
    valid_windows = {}
    if M.is_window_valid(win.winid) then
      table.insert(valid_windows, win)
```

and then at the end of the loop you can assign:
```lua
  floating_windows = valid_windows
```
to get rid of any invalid windows you may have found along the way.

Reply:
Yes, thank you for the comment. I forgot the fact that `table.remove()` could cause bugs in a loop.